pkcs11/gkm/gkm-data-der.c

```diff
@@ -1075,6 +1196,110 @@ gkm_data_der_write_private_key_dsa (gcry_sexp_t s_key)
 }
 
 GBytes *
+gkm_data_der_write_public_key_ecdsa (gcry_sexp_t s_key)
+{
+	GNode *asn = NULL, *named_curve;
+	gcry_mpi_t d = NULL;
+	GBytes *result = NULL, *q = NULL;
+	gchar *q_data = NULL;
+	GQuark oid;
+	gchar *curve = NULL;
+	gsize q_size;
+
+	init_quarks ();
+
+	asn = egg_asn1x_create (pk_asn1_tab, "ECPublicKey");
+	g_return_val_if_fail (asn, NULL);
+
+	if (!gkm_sexp_extract_buffer (s_key, &q_data, &q_size, "ecdsa", "q", NULL) ||
+	    !gkm_sexp_extract_string (s_key, &curve, "ecdsa", "curve", NULL))
+		goto done;
+
+	oid = gkm_data_der_curve_to_oid (curve);
+	g_free (curve);
+	if (oid == -1)
+		goto done;
+
+	q = g_bytes_new_take (q_data, q_size);
+	if (q == NULL)
+		goto done;
+
+	named_curve = egg_asn1x_node (asn, "parameters", "namedCurve", NULL);
+
+	if (!gkm_data_asn1_write_bit_string (egg_asn1x_node (asn, "q", NULL), q, q_size*8) || /*XXX*/
```

Comment:
We are encoding bit string assuming it has multiply of 8 bits. But we got it from S-Exp which represents it as octets (hopefully reasonably padded). This branch is covered in only one test case only for 256 b EC, which didn't give me complete confidence in this code. But better comment should be pressent. I will add it (preferably with some test to make sure it works).